snowpack/src/hmr-server-engine.ts

```diff
@@ -110,9 +118,33 @@ export class EsmHmrEngine {
   }
 
   broadcastMessage(data: object) {
+    if (this.liveReloadDelayMs > 0) {
+      if (this.currentBatchTimeout) {
+        clearTimeout(this.currentBatchTimeout);
+      }
+      this.currentBatch.push(data);
+      this.currentBatchTimeout = setTimeout(() => this.broadcastBatch(), this.liveReloadDelayMs);
+    } else {
+      this.internalBroadcastMessage([data]);
+    }
+  }
+
+  broadcastBatch() {
+    if (this.currentBatchTimeout) {
+      clearTimeout(this.currentBatchTimeout);
+    }
+    if (this.currentBatch.length > 0) {
+      this.internalBroadcastMessage(this.currentBatch);
+      this.currentBatch = [];
+    }
+  }
+
+  private internalBroadcastMessage(dataArr: object[]) {
```

Comment:
From a quick glance, it's hard to tell how `internalBroadcastMessage` & `broadcastMessage` are different. I'd say keep this function private, rename to `sendMessages`, and add a quick JSDoc note above about how this is different than the public `broadcastMessage`.

Reply:
also nit: `dataArr` would benefit from being less generic. What about `messageBatch`?